lib/src/components/NxForm/stateful/__tests__/NxStatefulForm.test.tsx

```diff
@@ -0,0 +1,459 @@
+            submitBtn = component.getByRole('button', { name: 'Submit' });
+
+        expect(additionalBtn).toBeInTheDocument();
+
+        // ensure Cancel and Submit are still there too
+        expect(submitBtn).toBeInTheDocument();
+        expect(cancelBtn).toBeInTheDocument();
+      });
+    });
+
+    describe('submit mask', function() {
+      it('renders no status element if submitMaskState is not defined', function() {
+        expect(within(renderEl()!).queryByRole('status')).not.toBeInTheDocument();
+        expect(within(renderEl({ submitMaskState: null })!).queryByRole('status')).not.toBeInTheDocument();
+      });
+
+      describe('when submitMaskState is false', function() {
+        it('renders a status of "Submitting…" when submitMaskMessage is not defined', function() {
+          const status = quickRender({ submitMaskState: false }).getByRole('status');
+          expect(status).toBeInTheDocument();
+          expect(status).toHaveTextContent('Submitting…');
+        });
+
+        it('renders the submitMaskMessage as the status text when it is defined', function() {
+          const status = quickRender({
+            submitMaskState: false,
+            submitMaskMessage: 'foo',
+            submitMaskSuccessMessage: 'bar'
+          }).getByRole('status');
+
+          expect(status).toBeInTheDocument();
+          expect(status).toHaveTextContent('foo');
+        });
+      });
+
+      describe('when submitMaskState is false', function() {
+        it('renders a status of "Success!" when submitMaskSuccessMessage is not defined', function() {
+          const status = quickRender({ submitMaskState: true }).getByRole('status');
+          expect(status).toBeInTheDocument();
+          expect(status).toHaveTextContent('Success!');
+        });
+
+        it('renders the submitMaskSuccessMessage as the status text when it is defined', function() {
+          const status = quickRender({
+            submitMaskState: true,
+            submitMaskMessage: 'foo',
+            submitMaskSuccessMessage: 'bar'
+          }).getByRole('status');
+
+          expect(status).toBeInTheDocument();
+          expect(status).toHaveTextContent('bar');
+        });
+      });
+    });
+
+    describe('validation errors', function() {
+      it('renders an alert when there are validation errors and no submitError', function() {
+        expect(within(renderEl()!).queryByRole('alert')).not.toBeInTheDocument();
+        expect(within(renderEl({ validationErrors: null })!).queryByRole('alert')).not.toBeInTheDocument();
+        expect(within(renderEl({ validationErrors: [] })!).queryByRole('alert')).not.toBeInTheDocument();
+        expect(within(renderEl({ validationErrors: 'foo' })!).getByRole('alert')).toBeInTheDocument();
+        expect(within(renderEl({ validationErrors: ['foo'] })!).getByRole('alert')).toBeInTheDocument();
+        expect(within(renderEl({ submitError: 'bar', validationErrors: 'foo' })!).getByRole('alert'))
```

Comment:
This one has `submitError: 'bar'`, should the alert still be rendered?

Reply:
> This one has submitError: 'bar', should the alert still be rendered?

Hmm this line isn't written right, thanks for catching that.  The idea is that when the submit alert is visible, the validation alert shouldn't be.  I'll adjust it to be more picky about the alert it queries, and fix the assertion.

Reply:
> Hmm this line isn't written right, thanks for catching that. The idea is that when the submit alert is visible, the validation alert shouldn't be. I'll adjust it to be more picky about the alert it queries, and fix the assertion.

b078df21f